desafio-05/cbcalves/c/v3/d5_v3.c

```diff
@@ -439,7 +445,12 @@ int main(int argc, char *argv[])
 
     // Inicialização do controle de threads
     // definindo numero de threads
-    unsigned threads_max = sysconf(_SC_NPROCESSORS_ONLN);
+    unsigned threads_max;
+    if (argc == 2) {
+        threads_max = sysconf(_SC_NPROCESSORS_ONLN);
+    } else {
+        threads_max = atoi(argv[1]);
```

Comment:
Qual o comportamento esperado? Na linha 430 o programa usa argv[1] como nome do arquivo, e aqui usa como número de threads? Eu imagino que o ideal seria `programa arquivo [num_threads]` (onde num_threads é opcional). É esse o comportamento esperado?

Reply:
Fala do Bernardino dia 6 de janeiro as 21:25 horário de Brasilia: "Coloca o 1º Parâmetro como sendo o número de Threads e o 2º o arquivo"
Meu ponto foi não inventar muito nisso, pois tem outros aprovados que fizeram o mesmo, acredito que o Bernardino tenha um script já adaptado para isso.

Você esqueceu de ver o if(argc==2) {...} else {...}

No C o argc sempre vai ser 1 ou mais, pois o primeiro parametro (argv[0])é a chamada ao programa. Então se argc== 2 ele entende que o argv[1] é o nome do arquivo para abrir e não tem mais parametros daí tenta estimar, perguntando ao sistema, quantos threads existem para usar. Se for argc >= 3 parametros (entrou no "else") ele pega o argv[2] como nome do arquivo a abrir e depois pega argv[1] como o número de threads para usar e ignora o resto dos parametros se existirem.